### DISCUSSION TITLE
Missing Module (or a Configuration Issue?)

### DISCUSSION THREAD

**Discussion by PhilRunninger** [mfussenegger/nvim-dap#850] [Q&A]
I'm working on a nvim-cmp completion source plugin. It's my first major effort in lua, and I'm running to a situation where a debugger would be really handy. I watched @tjdevries' very informative [nvim-dap video](https://www.youtube.com/watch?v=0moS8UHupGc), and tracked down the [jbyuki/one-small-step-for-vimkind adapter](https://github.com/jbyuki/one-small-step-for-vimkind) for Neovim lua scripts. I believe I've set it up correctly, like so:
```lua
local dap = require("dap")

dap.configurations.lua = {
  {
    type = 'nlua',
    request = 'attach',
    name = "Attach to running Neovim instance",
  }
}

dap.adapters.nlua = function(callback, config)
  callback({ type = 'server', host = config.host or "127.0.0.1", port = config.port or 8086 })
end

vim.api.nvim_set_keymap('n', '<F7>', [[:lua require("osv").launch({port = 8086})<CR>]], { noremap = true })

vim.api.nvim_set_keymap('n', '<F5>', [[:lua require("dap").continue()<CR>]], { noremap = true })
vim.api.nvim_set_keymap('n', '<F9>', [[:lua require("dap").toggle_breakpoint()<CR>]], { noremap = true })
vim.api.nvim_set_keymap('n', '<F10>', [[:lua require("dap").step_over()<CR>]], { noremap = true })
vim.api.nvim_set_keymap('n', '<F11>', [[:lua require("dap").step_into()<CR>]], { noremap = true })
vim.api.nvim_set_keymap('n', '<F12>', [[:lua require("dap.ui.widgets").hover()<CR>]], { noremap = true })
```
In the debuggee, I successfully launch the adapter's server. In my script, I can set a breakpoint, but when I run `continue()`, I get the error message below. I'm not sure if it's a bug or a configuration issue, but I thought I'd start a discussion here to get some answers.
```
lua: stdin:1: module 'bit' not found:
        no field package.preload['bit']
        no file '/usr/local/share/lua/5.4/bit.lua'
        no file '/usr/local/share/lua/5.4/bit/init.lua'
        no file '/usr/local/lib/lua/5.4/bit.lua'
        no file '/usr/local/lib/lua/5.4/bit/init.lua'
        no file './bit.lua'
        no file './bit/init.lua'
        no file '/usr/local/lib/lua/5.4/bit.so'
        no file '/usr/local/lib/lua/5.4/loadall.so'
        no file './bit.so'
stack traceback:
        [C]: in function 'require'
        stdin:1: in main chunk
        [C]: in ?
```
I'm working on a Macbook, and this is the result of running the `:version` command:
```
NVIM v0.8.0-1210-gd367ed9b2
Build type: Release
LuaJIT 2.1.0-beta3
Compiled by runner@Mac-1664551586262.local

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/share/nvim"

Run :checkhealth for more info
```

**Comment by PhilRunninger** [comment]
I'm digging into this more, and starting to figure it out a little at a time. Regarding the `bit` module  in the error message, my guess is `stdin` is reading my script, in which `local bit=require('bit')` is line 1. That begs the question: why is the script able to find the `bit` module when I'm not debugging it? Is this a question to be asked in the [jbyuki/one-small-step-for-vimkind adapter](https://github.com/jbyuki/one-small-step-for-vimkind) repo?

**Reply by jbyuki** [reply]
The module `bit` is only available in luajit. It's part of the "standard library" of luajit among other things. A recompile of neovim with luajit (5.1) as an internal interpreter instead of lua 5.4 would solve this issue.
